test/routing_test.rb

```diff
@@ -1452,4 +1452,22 @@ def authorize(username, password)
     end
     get '/users/1/status'
   end
+
+  it 'treats routes with and without trailing slashes differently' do
+    mock_app do
+      get '/foo' do
+        'Foo'
+      end
+
+      get '/foo/' do
+        'Foo with a slash'
+      end
+    end
+
+    get '/foo'
+    assert_equal 'Foo', body
```

Comment:
Hi @kgrz! I added the negative test case. But the test suite is failing (on master as well). I can see a comment [here](https://github.com/sinatra/sinatra/commit/36e5b26d35da68a8dd67951e27b424e5b60099f8#diff-fad785c2d32bb9236212277754d7c47dR238) but not sure what it means. Can you or @zzak  please verify this?